src/sentry/utils/snuba.py

```diff
@@ -1236,7 +1251,36 @@ def _bulk_snuba_query(snuba_requests: Sequence[SnubaRequest]) -> ResultSet:
                 if body.get("error"):
                     error = body["error"]
                     if response.status == 429:
+                        if options.get("issues.use-snuba-error-data"):
+                            if "stats" not in error:
+                                # Should not hit this - snuba always returns stats when there is an error
+                                raise RateLimitExceeded(error["message"])
+
+                            quota_allowance_summary = error["stats"]["quota_allowance"]["details"][
+                                "summary"
+                            ]
+                            rejected_by = quota_allowance_summary["rejected_by"]
+                            throttled_by = quota_allowance_summary["throttled_by"]
+
+                            if rejected_by:
+                                raise RateLimitExceeded(
+                                    error["message"],
+                                    error_type="rejected_by",
+                                    quota_used=rejected_by["quota_used"],
+                                    rejection_threshold=rejected_by["rejection_threshold"],
+                                    suggestion=rejected_by["suggestion"],
+                                )
```

Comment:
I think looking at an actual issue with this info will be more instructive as to what you want to log:

<img width="701" height="399" alt="image" src="https://github.com/user-attachments/assets/a5066374-8be7-49ff-a0f7-82a418ccace2" />

`error_type='rejected_by'` doesn't really tell you anything. Also `quota_used` doesn't tell you anything without the `quota_unit` which can be bytes scanned or concurrent queries.

Reply:
what you would want to log is the rejecting and throttling policies